test/e2e/knative_serving_test.go

```diff
@@ -29,7 +29,11 @@ const (
 
 func TestKnativeServing(t *testing.T) {
 	caCtx := test.SetupClusterAdmin(t)
-
+	route, err := setupMetricsRoute(caCtx, "serving")
+	if err != nil {
+		t.Fatal("Failed to setup operator metrics route", err)
+	}
+	metricsURL := "http://" + route.Spec.Host + route.Spec.Path
```

Comment:
Why not move this setup into the test that needs it?

Reply:
Because this is needed later on, I tried to avoid passing route (I just need a url) as a parameter to `verifyHealthStatusMetric`. I could though just to reduce code in that function. Is that what you mean?

Reply:
Where is it needed?

Reply:
A few lines bellow: `verifyHealthStatusMetric(caCtx, metricsURL, "eventing_status", 1)`.

Reply:
Right, but why aren't we having the entire route creation in the respective subtest?

Reply:
Note that my comment is on lines 32-36, not just the one line.

Reply:
Because my thought is that I would like to expand this to more subtests (eg. check health status when we uninstall, initially I had it this way). So I considered it a top level setup. I can move it for now in the subtest ok?

Reply:
We can leave it as-is too I guess, not a strong opinion here.